plugins/derive/derive.ml

```diff
@@ -8,14 +8,18 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
-open Context
 open Context.Named.Declaration
+open Vernacentries.DefAttributes
 
 (** [start_deriving f suchthat lemma] starts a proof of [suchthat]
     (which can contain references to [f]) in the context extended by
     [f:=?x]. When the proof ends, [f] is defined as the value of [?x]
     and [lemma] as the proof. *)
-let start_deriving f suchthat name : Declare.Proof.t =
+let start_deriving ~atts CAst.{v=f;loc} suchthat name : Declare.Proof.t =
+
+  let scope, _local, poly, program_mode, user_warns, typing_flags, using, clearbody =
```

Comment:
clearbody is not useful as we're non Discharge
unless you want to add some sort of `Let Derive` I guess

Reply:
I would say that the question is the same as for e.g. `Definition`. It supports the `clearbody` flag even if it does not apply.